torchtext/experimental/functional.py

```diff
@@ -1,4 +1,29 @@
 import torch
+import torchtext
+from typing import List, Tuple
+
+
+def regex_tokenizer(line: str, patterns_list: List[Tuple[str, str]]) -> List[str]:
+    r"""Regex tokenizer for a string sentence.
+
+    Steps include
+    - lowercasing
+    - applying all regex replacements defines in patterns_list
+
+    Args:
+        line (str): a line of text to tokenize.
+        patterns_list (List[Tuple[str, str]]): a list of tuples (ordered pairs) which contain the regex pattern string as the first element and the replacement string as the second element.
+
+    Returns:
+        tokens (List[str]): a list of tokens after splitting on whitespace.
+    """
+
+    line = line.lower()
+    regex = torch.classes.torchtext.Regex('')
```

Comment:
I'd guess that creating this object is example, because the regex needs to be compiled.

Reply:
Moved this to a class attribute instead that's being passed in